### PR TITLE
tests: add TEST_QEMU_ONLY variable to run only tests where qemu is mandatory, TEST_PREFER_NSPAWN to run under nspawn unless unsupported, fix some testsuite issues on Debian

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

**Comment by bluca** [comment]
autopkgtest suite PR to run most tests under nspawn and split qemu tests under a different suite: https://salsa.debian.org/systemd-team/systemd/-/merge_requests/114

**Comment by bluca** [comment]
With these changes and the upstream-ci PR merged, now the bionic-* is roughly twice as fast - 1h30m for the amd64, 2h for the i386 one. Hopefully should be the end of the frequent timeouts. PTAL!

**Comment by mbiebl** [comment]
I can confirm the nice speed up. I'll leave it up to someone with more experience with the test suite, like @keszybz , to review and merge it.

File: test/run-integration-tests.sh [hunk 1]
```diff
@@ -10,8 +10,6 @@ fi
 args_no_clean=$(sed -r 's/\bclean.*\b//g' <<<$args)
 do_clean=$( [ "$args" = "$args_no_clean" ]; echo $? )
 
-ninja -C "$BUILD_DIR"
```
Comment (keszybz) [inline test/run-integration-tests.sh:13]: Yeah. We should have never done that.

**Review by keszybz — COMMENTED** [review]
LGTM.

>  1h30m for the amd64, 2h for the i386 one. 

Blink of an eye ;)

**Comment by bluca** [comment]
> LGTM.
> 
> > 1h30m for the amd64, 2h for the i386 one.
> 
> Blink of an eye ;)

Working on a few more things which should gain a whopping few extra minutes :-D

**Pull request merged**